converter/stitching/stitcher.go

```diff
@@ -124,41 +127,51 @@ func (s *stitcher) stitchFlow(flow input.Flow) error {
 		if s.shouldMerge(&newSessAgg, &oldSessAgg) {
 			matchFound = true
 
-			//merge the session aggregates (adding data counts etc)
-			err = newSessAgg.Merge(&oldSessAgg)
-			if err != nil {
-				return errors.Wrapf(err, "cannot merge session\n%+v\nwith\n%+v", &newSessAgg, &oldSessAgg)
+			f_FlowStartMilliseconds := session.FlowStartMilliseconds(&newSessAgg)
+			nextMatch_FlowStartMilliseconds := session.FlowStartMilliseconds(&oldSessAgg)
+			f_FlowEndMilliseconds := session.FlowEndMilliseconds(&newSessAgg)
+			nextMatch_FlowEndMilliseconds := session.FlowEndMilliseconds(&newSessAgg)
+
+			var diff1 = f_FlowEndMilliseconds - nextMatch_FlowEndMilliseconds
+			if diff1 < 0 {
+				diff1 *= -1
 			}
-			//if both sides of the session have been filled, write it out
-			if newSessAgg.FilledFromSourceA && newSessAgg.FilledFromSourceB {
-				err := s.matcher.Remove(&oldSessAgg)
-				if err != nil {
-					return errors.Wrap(err, "could not remove old session aggregate")
-				}
-				s.sessionsOut <- &newSessAgg
-			} else {
-				//otherwise update the database
-				newSessAgg.MatcherID = oldSessAgg.MatcherID
-				err := s.matcher.Update(&newSessAgg)
-				if err != nil {
-					return errors.Wrap(err, "could not update existing session aggregate")
-				}
+			var diff2 = f_FlowStartMilliseconds - nextMatch_FlowStartMilliseconds
+			if diff2 < 0 {
+				diff2 *= -1
 			}
-		}
-	}
 
-	//if there's an error other than not found, return it up
-	if oldSessAggIter.Err() != nil /*&& oldSessAggIter.Err() != mgo.ErrNotFound*/ {
-		return errors.Wrap(oldSessAggIter.Err(), "could not find all matching session aggregates")
+			newMatchCost := diff1 + diff2
+
+			if newMatchCost < matchCost {
+				matchAgg = true
+			}
+		}
 	}
 
-	//no matching unstitched session found
-	if !matchFound {
-		err := s.matcher.Insert(&newSessAgg)
+	if matchAgg {
```

Comment:
since `matchAgg` will hold the result of our search, this line should be `if matchFound`